### DISCUSSION TITLE
Questions about the release process

### DISCUSSION THREAD

**Discussion by rndmh3ro** [ansible-collections/community.grafana#338] [General]
Hey @rrey,

i have some questions about the release process. :)

* I can see the general documentation [here](https://github.com/ansible-collections/community.grafana#publishing-new-versions). Do you have a link to some kind of further documentation or code on how that works?
* Do you generally do the releases? Can @Nemental or I support you here?

**Comment by rrey** [comment]
I'll try to write down something for you during the weekend.

**Reply by rrey** [reply]
So, basically the release process uses [antsibull-changelog](https://github.com/ansible-community/antsibull-changelog) and this is why the changelog fragments are so important.

You can generate the changelog for a release with the following command:
```
$ antsibull-changelog release
```

The tool parses the new fragment since last release and generates new changelog.
You are supposed to increment the `version` field in `galaxy.yml` prior running the command so that the changelog has the proper version.
But I have to admit that I'm using the changelog to see what the release contains as minor/major/bugfixes changes to actually know what is the proper semver increment.

So what I usually did in the past is:
* Generate the changelog
* check the changelog and determine new semver increment
* reset change, update `galaxy.yml` with proper version and re-generate the changelog.

Then, commit everything and create a PR like [this one](https://github.com/ansible-collections/community.grafana/pull/320/files).
After the merge I create the Github release:
* tag is the semver in format `X.Y.Z`
* release name is `vX.Y.Z`
* release description is the changelog section for the release.

Part of this can most certainly be automated.
Once the release is created, the release process to Galaxy is managed by the Ansible Core team, privilege of being a collection from the community namespace.

Thanks for all the work you did, that rocks.
